### PR TITLE
Remove nvidia-cg-toolkit as input for retroarch.

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [x] OS X
  - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

see #15707

### CODE REVIEW

**Comment by joachifm** [comment]
Makes sense to me. What remains to be done?

**Comment by matthewbauer** [comment]
I'm trying to get it to work on Darwin still but it might be more difficult than I was expecting.

**Comment by matthewbauer** [comment]
This should be ready if all tests pass.

**Comment by matthewbauer** [comment]
This should be ready to merge.

File: pkgs/misc/emulators/retroarch/default.nix [hunk 1]
```diff
@@ -44,11 +51,11 @@ stdenv.mkDerivation rec {
...
   enableParallelBuilding = true;
 
-  meta = with stdenv.lib; {
+  meta = {
     homepage = http://libretro.org/;
     description = "Multi-platform emulator frontend for libretro cores";
     license = licenses.gpl3;
-    platforms = stdenv.lib.platforms.linux;
-    maintainers = with maintainers; [ MP2E edwtjo ];
+    platforms = with platforms; [ linux darwin ];
```
Comment (joachifm) [inline pkgs/misc/emulators/retroarch/default.nix]: Both `linux` and `darwin` are lists, so this should be `linux ++ darwin`, strictly speaking.

**Comment by matthewbauer** [comment]
@joachifm I'm just going to put it as "platforms.all" because there's no reason it shouldn't work on other platforms but I've only got it tested on Linux and Darwin.

**Pull request merged**